src/main/java/com/foundation/view/AdvancedPanel.java

```diff
@@ -0,0 +1,271 @@
+/*
+ *  @(#)AdvancedPanel.java Copyright (c) 2018 Jalasoft.
+ *  2643 Av Melchor Perez de Olguin, Colquiri Sud, Cochabamba, Bolivia.
+ *  All rights reserved.
+ *
+ *  This software is the confidential and proprietary information of
+ *  Jalasoft, ("Confidential Information").  You shall not
+ *  disclose such Confidential Information and shall use it only in
+ *  accordance with the terms of the license agreement you entered into
+ *  with Jalasoft.
+ *
+ */
+
+package com.foundation.view;
+
+import javax.swing.*;
+import javax.swing.border.Border;
+import java.awt.Dimension;
+import java.awt.GridBagLayout;
+import java.awt.GridBagConstraints;
+import java.awt.Insets;
+
+/*
+ * Panel class setting with grid definition
+ *
+ * @Author Ruben Mendoza
+ * @Version 1.0
+ */
+
+public class AdvancedPanel extends JPanel {
+    private JLabel extLabel, visibilityLabel, ownerLabel, fileSizeLabel, dateCreatedLabel, dateModifiedLabel, dateAccessedLabel;
+    private JTextField ownerField, fileSizeField;
+    private FormListener formListener;
+    private JComboBox extList, visibilityList, fileCompareList, fileSizeList, dateCreatedPicker, dateModifiedPicker, dateAccessedPicker;
+
+    public AdvancedPanel(){
+        // check the current preferred Size and set a new one
+        Dimension dim = getPreferredSize();
```

Comment:
use methods like setting or init in order to not have in the constructor all the code, separate the functionality of the code in methods.

Reply:
This class will not be used at this point; its content will change and the fixes will be included when it will be implemented